tools/test-framework/src/chain/ext/ica.rs

```diff
@@ -92,7 +94,54 @@ pub fn register_interchain_account<Chain: ChainHandle, Counterparty: ChainHandle
 
     for event in events.iter() {
         if let IbcEvent::OpenInitChannel(open_init) = &event.event {
-            let channel_id = open_init.channel_id.clone().ok_or(()).map_err(|_| Error::generic(eyre!("channel_id is empty in the event response after sending  MsgRegisterInterchainAccount")))?;
+            let channel_id = open_init.channel_id.clone().ok_or(()).map_err(|_| Error::generic(eyre!("channel_id is empty in the event response after sending MsgRegisterInterchainAccount")))?;
+            return Ok((
+                wallet,
+                TaggedChannelId::new(channel_id),
+                TaggedPortId::new(open_init.port_id.clone()),
+            ));
+        }
+    }
+
+    Err(Error::generic(eyre!("could not retrieve an OpenInitChannel event response after sending MsgRegisterInterchainAccount")))
+}
+
+#[allow(clippy::type_complexity)]
+pub fn register_ordered_interchain_account<Chain: ChainHandle, Counterparty: ChainHandle>(
+    chain: &MonoTagged<Chain, FullNode>,
+    handle: &Chain,
+    connection: &ConnectedConnection<Chain, Counterparty>,
+) -> Result<
+    (
+        MonoTagged<Chain, Wallet>,
+        TaggedChannelId<Chain, Counterparty>,
+        TaggedPortId<Chain, Counterparty>,
+    ),
+    Error,
+> {
+    let wallet = chain.wallets().relayer().cloned();
+
+    let owner = handle.get_signer()?;
+
+    let version_str = format!("{{\"version\":\"ics27-1\",\"encoding\":\"proto3\",\"tx_type\":\"sdk_multi_msg\",\"controller_connection_id\":\"{}\",\"host_connection_id\":\"{}\"}}", connection.connection_id_a.0, connection.connection_id_b.0);
+    let msg = MsgRegisterInterchainAccount {
+        owner,
+        connection_id: connection.connection_id_a.0.clone(),
+        version: Version::new(version_str),
+        ordering: Ordering::Ordered,
+    };
+
+    let msg_any = msg.to_any();
+
+    let tm = TrackedMsgs::new_static(vec![msg_any], "RegisterInterchainAccount");
+
+    let events = handle
+        .send_messages_and_wait_commit(tm)
+        .map_err(Error::relayer)?;
+
+    for event in events.iter() {
+        if let IbcEvent::OpenInitChannel(open_init) = &event.event {
+            let channel_id = open_init.channel_id.clone().ok_or(()).map_err(|_| Error::generic(eyre!("channel_id is empty in the event response after sending MsgRegisterInterchainAccount")))?;
```

Comment:
This seems like an overly long line, would have expected rustfmt to shorten it.

Reply:
This seems to be ignored by `cargo fmt --all`, not sure why

Reply:
Manually done in 1f6d019b067c98f4b0e7938b1f940d6496920218